spec/index.html

```diff
@@ -1450,6 +1470,233 @@ <h2>Generalizations of RDF Triples, Graphs, and Datasets</h2>
 </section>
 
 
+<section id="section-classic-full-interop" class="informative">
+  <h2>Interoperability between RDF [=Classic=] and RDF [=Full=]</h2>
+
+  <p class=issue>Should we make this section normative?</p>
+
+  <p>This section provides transformations between [=Full=] [=RDF graphs=] (resp. [=RDF datasets=]) and [=Classic=] [=RDF graphs=] (resp. [=RDF datasets=]),
+  to provide some level of interoperability between the different classes of <a href="#conformance">Conformance</a>.
+
+  <p class=issue>Should we go even further and aim to provide interoperability between <em>RDF 1.1</em> and RDF 1.2 [=Full=]?</p>
+
+  <p class=issue>AT RISK: the working group may decide to replace the terms `rdf:TripleTerm`, `rdf:ttSubject`, `rdf:ttPredicate` and `rdf:ttObject` used in this section by other terms, possibly in a different namespace.</p>
+
+  <p>These transformation are designed to be:</p>
+
+  <dl>
+    <dt>Information preserving</dt>
+    <dd>It must be possible to reconstruct the input graph (resp. dataset) from the output graph (resp. dataset).
+      Note that, on the other hand, these transformations are not designed to be semantics preserving:
+      the output graph is not semantically [=equivalent=] to the input graph,
+      at least not in the entailment regimes defined in [[RDF12-SEMANTICS]].
+    </dd>
+    <dt>Idempotent</dt>
+    <dd>Applying a transformation several time to a graph (resp. dataset) should have the same effect as applying it once.
+      Moreover, [=classicizing=] a graph (resp. dataset) that is already complying with RDF [=Classic=] (i.e. containing no [=triple term=]) must result in the same graph (resp. dataset).
+    </dd>
+    <dt>Universal</dt>
+    <dd>It should be possible to transform any [=Full=] graph (resp. dataset) to a [=Classic=] graph (resp. dataset) using this method.
+      There is actually <a href="#section-classicize-caveat">a minor caveat</a> to this property.
+    </dd>
+  </dl>
+
+  <section id="section-classicize-definition">
+    <h2>From [=Full=] to [=Classic=]</h2>
+
+    <p>
+      Encoding an [=RDF graph=] to ensure that it is consumable by an RDF [=Classic=] implementation is called <dfn data-lt="classicize|classicized">classicizing</dfn> it.
+      It consists, while the graph has a [=triple term=] <var>tt</var> in its [=constituent terms=], to mint a fresh [=blank node=] <var>b</var>
+      (i.e. a blank node not in use in the graph), and replace <var>tt</var> with <var>b</var> in all the triples of the graph having <var>tt</var> in their [=constituents=].
+      Then the following triples are added to the graph (where <var>s</var>, <var>p</var> and <var>o</var> are respectively the [=subject=], [=predicate=] and [=object=] of <var>tt</var>):
```

Comment:
This section is written considering that graphs are mutable (add or remove something to a graph), while we define a graph as being static:

> The RDF data model is atemporal: [RDF graphs](https://raw.githack.com/w3c/rdf-concepts/refs/heads/action-129/spec/index.html#dfn-rdf-graph) are static snapshots of information.

Would it not be cleaner to describe the creation of a new graph after applying such transformations to make these operations more functional?

Reply:
> This section is written considering that graphs are mutable (add or remove something to a graph), while we define a graph as being static

Yes, I agree that could be confusing, but there are precedents of speaking of "replacing" nodes in a graph (sec 3.7 and 3.8). And I think that this is a concise and intuitive way to convey what the transformation is.

Note that the algorithm, on the other hand assumes immutable graphs and produces a new graph from scratch.